src/backend/distributed/citus--7.3-2--7.3-3.sql

```diff
@@ -0,0 +1,75 @@
+/* citus--7.3-2--7.3-3 */
+
+/*****************************************************************************
+ * Citus json aggregate helpers
+ *****************************************************************************/
+
+CREATE FUNCTION pg_catalog.citus_jsonb_concatenate(state jsonb, val jsonb)
+	RETURNS jsonb
+	LANGUAGE SQL
+AS $function$
+	SELECT CASE 
+		WHEN val IS NULL THEN state
+		WHEN jsonb_typeof(state) = 'null' THEN val
+		ELSE state || val
+	END;
+$function$;
+
+CREATE FUNCTION pg_catalog.citus_jsonb_concatenate_final(state jsonb)
+	RETURNS jsonb
+	LANGUAGE SQL
+AS $function$
+	SELECT CASE WHEN jsonb_typeof(state) = 'null' THEN NULL ELSE state END;
+$function$;
+
+CREATE FUNCTION pg_catalog.citus_json_concatenate(state json, val json)
+	RETURNS json
+	LANGUAGE SQL
+AS $function$
+	SELECT CASE 
+		WHEN val IS NULL THEN state
+		WHEN json_typeof(state) = 'null' THEN val
+		WHEN json_typeof(state) = 'object' THEN
+ 			(SELECT json_object_agg(key, value) FROM (
```

Comment:
This is not so elegant but postgres does not have `jsonb_concat` equivalent for `json`.

Reply:
It might be simpler to just convert to JSONB and back, which wouldn't be significantly more expensive since these functions pay the parsing/deparsing cost anyway.

Reply:
We would lose consistent behaviour with vanilla `json_object_agg` since converting to `jsonb` will lose duplicated keys and ordering.